extensions/wikia/Insights/InsightsHooks.class.php

```diff
@@ -130,4 +130,14 @@ public static function onAfterUpdateSpecialPages( $queryPage ) {
 
 		return true;
 	}
+
+	public static function onTemplateClassified( $pageId, $title, $templateType ) {
+		if ( !RecognizedTemplatesProvider::isUnrecognized( $templateType ) ) {
+			$model = new InsightsTemplatesWithoutTypeModel();
+			$model->removeFixedItem( TemplatesWithoutTypePage::TEMPLATES_WITHOUT_TYPE_TYPE, $title );
+			$model->purgeInsightsCache();
```

Comment:
Why do we need to purge the cache if the item is selectively removed in the line above?